lib/matplotlib/backend_bases.py

```diff
@@ -1595,6 +1595,73 @@ def __init__(self, figure):
         self.toolbar = None  # NavigationToolbar2 will set me
         self._is_idle_drawing = False
 
+    def _repr_html_(self):
+        if not self.figure.axes and not self.figure.lines:
+            return
+
+        dpi = self.figure.dpi
+        is_retina = False
+
+        import IPython
+        ip = IPython.get_ipython()
+        ib_list = [c for c in ip.configurables 
+                    if 'InlineBackend' in type(c).__name__]
+
+        # having an 'inline' backend doesn't mean '%matplotlib inline' has been run
+        # Running %matplotlib inline runs pylabtools.configure_inline_support
+        # which appends the InlineBackend to the list of configurables
+        if get_backend() == 'module://ipykernel.pylab.backend_inline' and ib_list:
+            ib = ib_list[0]
+            bbox_inches = ib.print_figure_kwargs['bbox_inches']
+            fmt = next(iter(ib.figure_formats))
+            if fmt == 'retina':
+                is_retina = True
+                dpi = dpi * 2
+                fmt = self.get_default_filetype()
+        else:
+            bbox_inches = 'tight' # how to let user choose self.figure.bbox_inches?
```

Comment:
This just should not be passed or set to None.  That IPython inline defaults to `'tight'` has caused a tremendous amount of confusion over the years.